person/src/com/person/core/model/BasePerson.java

```diff
@@ -0,0 +1,178 @@
+package com.person.core.model;
+
+import com.person.core.exceptions.ValidationException;
+import com.person.core.service.PersonAction;
+import com.person.utils.Validate;
+
+public abstract class BasePerson implements PersonAction {
```

Comment:
@tatevikb why are you implementing Person Action it is service , model is your pure object , model should not implement service